src/theory/sep/theory_sep.cpp

```diff
@@ -184,76 +171,95 @@ void TheorySep::computeCareGraph() {
-
+  if (d_type_ref.isNull())
+  {
+    return;
+  }
   NodeManager* nm = NodeManager::currentNM();
   std::vector< Node > sep_children;
   Node m_neq;
   Node m_heap;
-  for( std::map< TypeNode, Node >::iterator it = d_base_label.begin(); it != d_base_label.end(); ++it ){
-    //should only be constructing for one heap type
-    Assert(m_heap.isNull());
-    Assert(d_loc_to_data_type.find(it->first) != d_loc_to_data_type.end());
-    Trace("sep-model") << "Model for heap, type = " << it->first << " with data type " << d_loc_to_data_type[it->first] << " : " << std::endl;
-    TypeNode data_type = d_loc_to_data_type[it->first];
-    computeLabelModel( it->second );
-    if( d_label_model[it->second].d_heap_locs_model.empty() ){
-      Trace("sep-model") << "  [empty]" << std::endl;
-    }else{
-      for( unsigned j=0; j<d_label_model[it->second].d_heap_locs_model.size(); j++ ){
-        Assert(d_label_model[it->second].d_heap_locs_model[j].getKind()
-               == kind::SET_SINGLETON);
-        std::vector< Node > pto_children;
-        Node l = d_label_model[it->second].d_heap_locs_model[j][0];
-        Assert(l.isConst());
-        pto_children.push_back( l );
-        Trace("sep-model") << " " << l << " -> ";
-        if( d_pto_model[l].isNull() ){
-          Trace("sep-model") << "_";
-          TypeEnumerator te_range( data_type );
-          if (d_env.isFiniteType(data_type))
-          {
-            pto_children.push_back( *te_range );
-          }else{
-            //must enumerate until we find one that is not explicitly pointed to
-            bool success = false;
-            Node cv;
-            do{
-              cv = *te_range;
-              if( std::find( d_heap_locs_nptos[l].begin(), d_heap_locs_nptos[l].end(), cv )==d_heap_locs_nptos[l].end() ){
-                success = true;
-              }else{
-                ++te_range;
-              }
-            }while( !success );
-            pto_children.push_back( cv );
-          }
+  // should only be constructing for one heap type
+  Assert(m_heap.isNull());
+  Trace("sep-model") << "Model for heap, type = " << d_type_ref
+                     << " with data type " << d_type_data << " : " << std::endl;
+  computeLabelModel(d_base_label);
+  HeapInfo& hm = d_label_model[d_base_label];
+  if (hm.d_heap_locs_model.empty())
+  {
+    Trace("sep-model") << "  [empty]" << std::endl;
+  }
+  else
+  {
+    for (const Node& hv : hm.d_heap_locs_model)
+    {
+      Assert(hv.getKind() == SET_SINGLETON);
+      std::vector<Node> pto_children;
+      Node l = hv[0];
+      Assert(l.isConst());
+      pto_children.push_back(l);
+      Trace("sep-model") << " " << l << " -> ";
+      if (d_pto_model[l].isNull())
+      {
+        Trace("sep-model") << "_";
+        TypeEnumerator te_range(d_type_data);
+        if (d_env.isFiniteType(d_type_data))
+        {
+          pto_children.push_back(*te_range);
         }else{
```

Comment:
Since most of the file has changed, I prefer to format the whole file, not only the changes.

Reply:
This would take a lot more effort, I'll do this separately.